src/main/java/uk/gov/ida/verifyserviceprovider/configuration/MatchingServiceAdapterConfiguration.java

```diff
@@ -51,11 +51,21 @@ public String getTrustStorePassword() {
         throw new NotImplementedException("MSA metadata is not signed, so no trust store available");
     }
 
-    @NotNull
-    @Valid
     @Override
     public URI getUri() {
-        return uri;
+        return metadataUri;
+    }
+
+    @NotNull
+    @Valid
+    public URI getMetadataUri() {
+        return metadataUri;
+    }
+
+    @NotNull
+    @Valid
+    public String getEntityId() {
+        return entityId;
     }
```

Comment:
It looks like if we move the `@NotNull` and `@Valid` annotations to the fields rather than the getters, we can have this work (giving the more useful names in error messages) without needing the extra getters, which would be a bit nicer.